src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1060,7 +1060,7 @@ FixedwingPositionControl::control_position(const Vector2f &curr_pos, const Vecto
 		/* making sure again that the correct thrust is used,
 		 * without depending on library calls for safety reasons.
 		   the pre-takeoff throttle and the idle throttle normally map to the same parameter. */
-		_att_sp.thrust = _parameters.throttle_idle;
+		_att_sp.thrust = _launchDetector.getThrottleIdle(_parameters.throttle_idle);
```

Comment:
Typically the pattern here is to pass the value you'd use in the non-launch detection case, and launch detector would return one or the other depending on if it's enabled.

Reply:
I believe this is what is happening. Or do I miss something? (at least this is what I intended)

[LaunchDetector.cpp#L128](https://github.com/tstastny/Firmware/blob/6cb7bc880f9c66b660d68bef96b730019427dbac/src/modules/fw_pos_control_l1/launchdetection/LaunchDetector.cpp#L128)

Reply:
My mistake, that looks correct.